### ISSUE TITLE
Binder repository for interactive GMT tutorials

### ISSUE THREAD

**Issue by meghanrjones** [GenericMappingTools/gmt#4627]
**Description of the desired feature**

A large part of the NASA proposal that we’re working on involves improving the GMT documentation, including tutorials. One goal is to make the tutorials reliably repeatable and fully reproducible. Remote files will be useful for this purpose, which are already used in the existing tutorial but not some other parts of the documentation. In addition, it would be great for people to be able to learn GMT, PyGMT, GMT.jl, and/or gmtmex without needing to go through the installation steps. My expectation is that Binder and JupyterLab environments would be an excellent way to ensure that the tutorials are reliably repeatable. The result would be similar to [try-gmt](https://github.com/GenericMappingTools/try-gmt). The binder-ready repos would complement static versions of the tutorials in the docs that users could follow in their local environments. Pinging @GenericMappingTools/core, @GenericMappingTools/gmt-contributors, @GenericMappingTools/python, and @GenericMappingTools/python-maintainers for opinions about whether this is worthwhile.

I am also to happy to chat in more detail about other documentation improvements included in the proposal, so please feel free to reach out. Slack is probably the easiest medium for a broader conversation about the proposal. 

**Are you willing to help implement and maintain this feature?** Yes

**Comment by seisman** [comment]
We already have a [bash demo](https://github.com/GenericMappingTools/try-gmt/blob/master/bash-demo.ipynb) in try-gmt. Could we just add more tutorials in the try-gmt binder, so that we don't have to maintain two similar environments?

**Comment by meghanrjones** [comment]
> We already have a [bash demo](https://github.com/GenericMappingTools/try-gmt/blob/master/bash-demo.ipynb) in try-gmt. Could we just add more tutorials in the try-gmt binder, so that we don't have to maintain two similar environments?

Yes, I might be easiest to create Jupyter Notebooks for the tutorials and include them in the try-gmt environment.

**Comment by GD-A** [comment]
May I (also) suggest an easy/clear access to the remote files. It's not always easy to find where or what the files are.
